spec/models/time_tracker_extension/time_locking_period_spec.rb

```diff
@@ -59,6 +59,26 @@ module TimeTrackerExtension
       it { should belong_to(:user) }
     end
 
+    describe "#approve!" do
+      let(:period) { create(:time_locking_period, approved: false) }
+
+      it "should set attribute approved as true" do
+        period.approve!
+        expect(period.reload.approved).to be_truthy
+      end
+
+      it "should launch job for sending reports" do
+        expect(TimeTrackerExtension::SendPeriodReportsJob).to receive(:perform_later).with(period)
```

Comment:
I think it's me but can you point to SendPeriodReportsJob class? 
I do not understand it and what is perform_later method doing.

Reply:
it's functionality of rails' active job, so I use this for generating reports in the background

Reply:
there is place for `SendPeriodReportsJob` class
https://github.com/x13enos/time_tracker_extension/blob/e65cd4ee60f2e691fe2195dec20ca7882fe14977/app/jobs/time_tracker_extension/send_period_reports_job.rb